common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java

```diff
@@ -199,35 +196,39 @@ public static void runTests(PerfStressTest<?>[] tests, boolean sync, int paralle
                 return String.format("%d\t\t%d\t\t%.2f", currentCompleted, totalCompleted, averageCompleted);
             }, true, true);
 
-        if (sync) {
-            ForkJoinPool forkJoinPool = new ForkJoinPool(parallel);
-            try {
-                forkJoinPool.submit(() -> {
-                    IntStream.range(0, parallel).parallel().forEach(i -> runLoop(tests[i], i, endNanoTime));
-                }).get();
-            } catch (InterruptedException | ExecutionException e) {
-                throw new RuntimeException(e);
+        try {
+            if (sync) {
+                ForkJoinPool forkJoinPool = new ForkJoinPool(parallel);
+                try {
+                    forkJoinPool.submit(() -> {
+                        IntStream.range(0, parallel).parallel().forEach(i -> runLoop(tests[i], i, endNanoTime));
+                    }).get();
+                } catch (InterruptedException | ExecutionException e) {
+                    throw new RuntimeException(e);
+                }
+            } else {
+                // Exceptions like OutOfMemoryError are handled differently by the default Reactor schedulers. Instead of terminating the
+                // Flux, the Flux will hang and the exception is only sent to the thread's uncaughtExceptionHandler and the Reactor
+                // Schedulers.onHandleError.  This handler ensures the perf framework will fail fast on any such exceptions.
+                Schedulers.onHandleError((t, e) -> {
+                    System.err.print(t + " threw exception: ");
+                    e.printStackTrace();
+                    System.exit(1);
+                });
+
+                Flux.range(0, parallel)
+                    .parallel()
+                    .runOn(Schedulers.boundedElastic())
+                    .flatMap(i -> runLoopAsync(tests[i], i, endNanoTime))
+                    .then()
+                    .block();
             }
-        } else {
-            // Exceptions like OutOfMemoryError are handled differently by the default Reactor schedulers. Instead of terminating the
-            // Flux, the Flux will hang and the exception is only sent to the thread's uncaughtExceptionHandler and the Reactor
-            // Schedulers.onHandleError.  This handler ensures the perf framework will fail fast on any such exceptions.
-            Schedulers.onHandleError((t, e) -> {
-                System.err.print(t + " threw exception: ");
-                e.printStackTrace();
-                System.exit(1);
-            });
-
-            Flux.range(0, parallel)
-                .parallel()
-                .runOn(Schedulers.boundedElastic())
-                .flatMap(i -> runLoopAsync(tests[i], i, endNanoTime))
-                .then()
-                .block();
+        } catch (Exception e) {
+            System.err.println("Error occurred running tests: " + System.lineSeparator() + e);
+        } finally {
+            progressStatus.dispose();
```

Comment:
Could we use try-with-resources?

Reply:
I think try-with-resources only works with AutoCloseable and not with Disposable.

Reply:
This is disposable. IIRC try-with-resources was for AutoCloseable?